### PR TITLE
added the l1_loss

### LINKED ISSUE #21066: l1_loss
- [l1_loss] #21068

### PR DESCRIPTION
**Pull request by sushmanthreddy**
close #21066

### CODE REVIEW

**Comment by sushmanthreddy** [comment]
@DavidAyankoso can u plss review th pull request ?

i have made required changes and it is good to merge

**Comment by sushmanthreddy** [comment]
@tomatillos I need help with this pr, I have impleme
Screenshot 2023-08-02 at 9 15 58 PM
nted it exactly in the correct way as they mentioned in the documentation.

but still some 
checks are not passing like module not found error..
My mac has a problem could test code in my local machine .

can u pls review it once ??and let me knew for any changes?

**Comment by tomatillos** [comment]
Same as your other PR, the docstring examples you provide don't match up with the code you wrote. Can you confirm that you have run this locally?

**Comment by sushmanthreddy** [comment]
@tomatillos I was working with different loss functions and I got confused and swapped the l1_loss with huber_loss
I will update it

**Comment by sushmanthreddy** [comment]
@tomatillos I have updated the docstrings , now I am getting the values as mentioned...but there are some precision errors..

**Comment by tomatillos** [comment]
Again, the relevant tests are not passing in the CI. If you need help setting up the tests let me know. We can't merge code that introduces more test failures! Thanks :)

**Comment by sushmanthreddy** [comment]
@tomatillos I have gone through the documentation clearly of  l1_loss. I have read the paper also, about l1_loss and understood the l1_loss implementation, and have implemented it correctly.
And it is really weird couldn't understand where ,I am going wrong..And the loss function implementation is also even simple..just we need to take absolute of `pred-true`..

i really don't understand ,where is wrong in my code . I have seen other implementation of cross entropy ,binary cross entrop and sparse cross entropy...they are also Implemented in the same way...can u pls at least point where might going wrong??

**Comment by sushmanthreddy** [comment]
@tomatillos this code has changed ... Now the tests are passing for tensorflow ,pytorch ,jax and numpy ...Only for paddle they are not passing that is due to some paddle backend issue of dtype bfloat16. More about the failure of this paddle issue can be found in [link](https://discord.com/channels/799879767196958751/1139168214723592264)


And for the stateful API has some issue ,it is not only failing for my code but also for other implementation which are being merged.

I have removed all lint errors for this pull request.so it can be merged now.

**Comment by tomatillos** [comment]
Unfortunately the CI is still showing that `test_l1_loss` fails for all backends, we can't merge this until the tests pass sadly. It's fine to leave this task if you can't get it working.

**Comment by sushmanthreddy** [comment]
@tomatillos 
Screenshot 2023-08-14 at 2 32 19 PM

**Comment by sushmanthreddy** [comment]
@tomatillos do u mean these checks??

**Comment by sushmanthreddy** [comment]
@tomatillos backend is failing because there is no implementation of l1_loss in backends yet..I think that is the reason ,for all failing checks in backend

**Comment by tomatillos** [comment]
The automatic checks (which you can see by going to checks -> intelligent-test-pr -> display_test_results -> `new_failures_introduced`) include test_l1_loss as a failure. The original task (https://github.com/unifyai/ivy/issues/3856) does say to implement in each backend API.

**Comment by sushmanthreddy** [comment]
> The automatic checks (which you can see by going to checks -> intelligent-test-pr -> display_test_results -> `new_failures_introduced`) include test_l1_loss as a failure. The original task (#3856) does say to implement in each backend API.

@tomatillos I really don't knew ,where iam going wrong...But I will implement the backend implementation for l1_loss..
But small doubt as mentioned here ,I have went there and saw the tests failing,when I ran test for them in my local machine then l1_loss is passing for all them except paddle that is because of bfloat16 issue.
Screenshot 2023-08-14 at 2 32 19 PM


and after seeing the tests results in checks -> intelligent-test-pr -> display_test_results -> new_failures_introduced
Screenshot 2023-08-14 at 3 27 54 PM


yeah, test results are failing but they are only failing for the log_poission_loss .And my implementation is about l1_loss and  l1_loss only for paddle they are failing. I think the issue with paddle is related to it converting bfloat16 to uint16 or soemthing.

And that bug persists in ivy lib.

**Comment by sushmanthreddy** [comment]
@tomatillos and backend implementation for l1_loss is there only for torch..and it is not there any other lib such as tensorflow,jax,numpy and paddle

**Comment by sushmanthreddy** [comment]
@tomatillos do we need to change anything??
here or it  is ready to merge??

**Comment by sushmanthreddy** [comment]
@tomatillos is there any update on this??
pls review it once ...your feedback is useful

**Comment by sushmanthreddy** [comment]
@tomatillos the stateful api support has been removed can ,u plss review it ?It has been three days..

File: ivy/functional/ivy/experimental/losses.py [hunk 1]
```diff
@@ -93,3 +94,60 @@ def log_poisson_loss(
...
+    ivy.array(0.5)
+    """
+    loss = ivy.abs(pred - true)
+
+    if reduction == "sum":
+        return ivy.sum(loss, out=out)
+    elif reduction == "mean":
+        return ivy.mean(loss, out=out)
+    else:
+        return loss
```
Comment (tomatillos) [inline ivy/functional/ivy/experimental/losses.py:153]: We should update `out` in place if it is passed, rather than returning loss

Comment (sushmanthreddy) [reply ivy/functional/ivy/experimental/losses.py]: @tomatillos it is done...checks after passing the in_place.
Screenshot from 2023-08-18 10-29-17

**Comment by sushmanthreddy** [comment]
@tomatillos it has been changed can u pls review it and spell those magic words 

"LGTM" ?

**Comment by sushmanthreddy** [comment]
@tomatillos unsupported datatypes are checked this way and updated it accordingly ,and crossverified with orginal implementation also.
Screenshot from 2023-08-20 02-43-48

File: ivy/data_classes/array/experimental/losses.py [hunk 2]
```diff
@@ -1,6 +1,50 @@
...
+        ivy.Array instance method variant of ivy.l1_loss. This method simply wraps the
+        function, and so the docstring for ivy.l1_loss also applies to this method with
+        minimal changes.
+
+        Parameters
+        ----------
+        self
+            input array.
+        target
+            input array containing the targeticted values.
```
Comment (tomatillos) [inline ivy/data_classes/array/experimental/losses.py:28]: Typo "targeticted", appears a few times in this file

**Comment by tomatillos** [comment]
Hi, it's best not to rush because there are still mistakes in both PRs you are requesting reviews on. This is taking up a lot of time and back and forth. The CI still says this function is failing the tests, e.g. (ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py::test_l1_loss,jax). Please take your time, rushing only introduces more mistakes and we spend more time going back and forth.

**Comment by sushmanthreddy** [comment]
sorry for the rush...i have created , i have changed targeted values spelling and tests are failing for jax because there is problem in function_testing.py someone had made changes to that file due to that there  is problem in compute gradients,and due to that tests have failed.
But today morning @vedpatwardhan has updated the code for  function_testing.py and tests will pass.My code hasn't been changed single bit except the spelling mistake of "targeted"..

Now tests will pass except for paddle and they are not passing for other implemented function also,here is the screenshot f it..
and that issue is bfloat16 in paddle
Screenshot from 2023-08-21 18-13-55

review it again and let me knew any changes are needed...

**Comment by sushmanthreddy** [comment]
@tomatillos sorry, i have made u rush to merge ,
